source/features/user-local-time.tsx

```diff
@@ -79,6 +79,35 @@ function parseOffset(date: string): number {
 	return (hours * 60) + (hours < 0 ? -minutes : minutes);
 }
 
+async function handleDatePromise({
```

Comment:
How about naming it something with display? Isn't that what this function does?

Reply:
It's just a random name I come up with, these code are so intertwined I have no idea what I'm doing.